src/shell/xdg/mod.rs

```diff
@@ -168,7 +168,7 @@ pub struct XdgPositioner(xdg_positioner::XdgPositioner);
 
 impl XdgPositioner {
     pub fn new(
-        wm_base: &impl ProvidesBoundGlobal<xdg_wm_base::XdgWmBase, 4>,
+        wm_base: &impl ProvidesBoundGlobal<xdg_wm_base::XdgWmBase, 5>,
```

Comment:
same about constants here.

Reply:
I'm less sure about making the change to the uses that require an interface instead of implementing it. Making it a constant will cause the documentation to refer to that constant instead of the version number, and the entire reason this number exists is so that you have to bump it if there is an API break.

Reply:
yes, but we have a situation where everyone forgot to bump anything and we have an API break.

Reply:
That's an argument for tying the bind and the trait impl together, but not an argument for also tying generic bounds to the constant.  The generic bounds being wrong would cause compile errors more directly.

Reply:
I mean, bumping the protocol version is already a breaking change most of the time, it's not like you can miss it that easily.

The only way is to not have this bounds at all, so we never have a breaking change.